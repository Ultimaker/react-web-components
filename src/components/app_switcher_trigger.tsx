import * as React from 'react';
import classNames from 'classnames';

export interface AppSwitcherTriggerProps {
    isAppSwitcherOpen: boolean;
}

export const AppSwitcherTrigger: React.FC<AppSwitcherTriggerProps> = ({
    isAppSwitcherOpen,
}) => {
    const classes = classNames('app_switcher_trigger', { 'app_switcher_trigger--open': isAppSwitcherOpen });

    return (
        <div className={classes}>
            <div className="grid-icon">
                <span className="layer layer--primary">
                    <span />
                </span>
                <span className="layer layer--secondary">
                    <span />
                </span>
                <span className="layer layer--tertiary">
                    <span />
                </span>
            </div>
        </div>
    );
};

AppSwitcherTrigger.displayName = 'AppSwitcherTrigger';

export default AppSwitcherTrigger;
