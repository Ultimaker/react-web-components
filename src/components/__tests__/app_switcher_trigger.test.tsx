// Copyright (c) 2018 Ultimaker B.V.
import * as React from 'react';
import { shallow } from 'enzyme';

// component
import { AppSwitcherTrigger, AppSwitcherTriggerProps } from '../app_switcher_trigger';

describe('The AppSwitcherTrigger component', () => {
    let props: AppSwitcherTriggerProps;
    let wrapper;

    beforeEach(() => {
        props = {
            isAppSwitcherOpen: false,
        };
        wrapper = shallow(<AppSwitcherTrigger {...props} />);
    });

    it('should render', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should change when app switcher is open', () => {
        wrapper.setProps({
            isAppSwitcherOpen: true,
        });
        expect(wrapper.find('.app_switcher_trigger').hasClass('app_switcher_trigger--open')).toBe(true);
    });
});
