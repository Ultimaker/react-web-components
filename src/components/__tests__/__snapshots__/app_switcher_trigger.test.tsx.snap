// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The AppSwitcherTrigger component should render 1`] = `
<div
  className="app_switcher_trigger"
>
  <div
    className="grid-icon"
  >
    <span
      className="layer layer--primary"
    >
      <span />
    </span>
    <span
      className="layer layer--secondary"
    >
      <span />
    </span>
    <span
      className="layer layer--tertiary"
    >
      <span />
    </span>
  </div>
</div>
`;
